drivers/sensor/iis2iclx/iis2iclx.c

```diff
@@ -641,17 +623,146 @@ static int iis2iclx_init(const struct device *dev)
 
 #ifdef CONFIG_IIS2ICLX_SENSORHUB
 	if (iis2iclx_shub_init(dev) < 0) {
-		LOG_ERR("failed to initialize external chip");
-		return -EIO;
+		LOG_INF("failed to initialize external chips");
+		data->shub_inited = false;
 	}
+	data->shub_inited = true;
 #endif
 
 	return 0;
 }
 
+#if DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 0
+#warning "IIS2ICLX driver enabled without any devices"
+#endif
+
+/*
+ * Device creation macro, shared by IIS2ICLX_DEFINE_SPI() and
+ * IIS2ICLX_DEFINE_I2C().
```

Comment:
At some point I feel like we should define some device driver conveniences for these "SPI2C" type devices to cut down on boilerplate.  But that is not for this PR.

Reply:
Defintely there is a lot of redundancy here.